draft-ietf-quic-tls.md

```diff
@@ -499,9 +499,12 @@ handshake, new data is requested from TLS after providing received data.
 ### Encryption Level Changes
 
 As keys at a given encryption level become available to TLS, TLS indicates to
-QUIC that reading or writing keys at that encryption level are available.  These
-events are not asynchronous; they always occur immediately after TLS is provided
-with new handshake bytes, or after TLS produces handshake bytes.
+QUIC that reading or writing keys at that encryption level are available.  While
+generating these keys, an endpoint SHOULD buffer received packets marked as
+protected by the keys being generated, and process them once those keys become
+available.  If the keys are generated asynchronously, an endpoint MAY continue
```

Comment:
I think we expect implementations to do this typically, unless there's a reason not to?

```suggestion
available.  If the keys are generated asynchronously, an endpoint SHOULD continue
```

Reply:
I don't think this needs a recommendation. It's an optimization, and endpoints can do as they choose.